driver/portfolio_runner.py

```diff
@@ -83,11 +83,12 @@ def compute_run_time(timeout, configs, pos):
     print("remaining time: {}".format(remaining_time))
     relative_time = configs[pos][0]
     remaining_relative_time = sum(config[0] for config in configs[pos:])
-    print("config {}: relative time {}, remaining {}".format(
-          pos, relative_time, remaining_relative_time))
     # For the last config we have relative_time == remaining_relative_time, so
     # we use all of the remaining time at the end.
-    return remaining_time * relative_time / remaining_relative_time
+    limit = int(remaining_time * relative_time / remaining_relative_time)
+    print("config {}: relative time {}, remaining {}, limit {}s".format(
+          pos, relative_time, remaining_relative_time, limit))
```

Comment:
Is it clear enough that `remaining` means "remaining *relative* time"? In the previous output you could argue that both times printed were relative, which hinted at this. Now we print three times, of which only two are relative. Also, I don't understand why we have a print statement here in the first place. Isn't this logging which should go through the logger?

Reply:
I would also prefer staying with the previous output with just the relative times. The actual time limit will be printed a few lines later in the log anyway as part of the call to the search component.

As for `print`/`logging`, the driver code is not consistent there. It uses `print` 23 times and `logging` 10 times. There is also a mix between `"%s" % x`, `"{}".format(x)`, and `f"{x}"`. I would prefer not to clean this up as part of this issue.

I suggest to just revert the print statement to its previous form.

Reply:
There is a difference between printing and logging, they are not the same concept. Not all existing uses of `print` are wrong. I had a brief look at this before commenting on this, and the first uses of `print` I found were correct. Looking at it again, there are indeed more uses that are wrong and should be replaced by logging, so I'm OK with your suggestion and don't mind leaving it alone here for now. It would only solve a small portion of the problem.

Using f-strings everywhere would also be nice, but I seed it on a lower level of importance because it is only code cleanup without affecting the user.